core/field_colour.js

```diff
@@ -197,47 +284,47 @@ Blockly.FieldColour.prototype.activateEyedropperInternal_ = function() {
-  Blockly.WidgetDiv.show(this, this.sourceBlock_.RTL,
-      Blockly.FieldColour.widgetDispose_);
-  // Create the palette using Closure.
-  var picker = new goog.ui.ColorPicker();
-  picker.setSize(this.columns_ || Blockly.FieldColour.COLUMNS);
-  picker.setColors(this.colours_ || Blockly.FieldColour.COLOURS);
+  Blockly.DropDownDiv.hideWithoutAnimation();
+  Blockly.DropDownDiv.clearContent();
+  var div = Blockly.DropDownDiv.getContentDiv();
 
-  // Position the palette to line up with the field.
-  // Record windowSize and scrollOffset before adding the palette.
-  var windowSize = goog.dom.getViewportSize();
-  var scrollOffset = goog.style.getViewportPageOffset(document);
-  var xy = this.getAbsoluteXY_();
-  var borderBBox = this.getScaledBBox_();
-  var div = Blockly.WidgetDiv.DIV;
-  picker.render(div);
-  picker.setSelectedColor(this.getValue());
-  // Record paletteSize after adding the palette.
-  var paletteSize = goog.style.getSize(picker.getElement());
+  var hueElements = this.createLabelDom_('Hue');
+  div.appendChild(hueElements[0]);
+  this.hueReadout_ = hueElements[1];
+  this.hueSlider_ = new goog.ui.Slider();
+  this.hueSlider_.setUnitIncrement(5);
+  this.hueSlider_.setMinimum(0);
+  this.hueSlider_.setMaximum(359);
+  this.hueSlider_.render(div);
 
-  // Flip the palette vertically if off the bottom.
-  if (xy.y + paletteSize.height + borderBBox.height >=
-      windowSize.height + scrollOffset.y) {
-    xy.y -= paletteSize.height - 1;
-  } else {
-    xy.y += borderBBox.height - 1;
-  }
-  if (this.sourceBlock_.RTL) {
-    xy.x += borderBBox.width;
-    xy.x -= paletteSize.width;
-    // Don't go offscreen left.
-    if (xy.x < scrollOffset.x) {
-      xy.x = scrollOffset.x;
-    }
-  } else {
-    // Don't go offscreen right.
-    if (xy.x > windowSize.width + scrollOffset.x - paletteSize.width) {
-      xy.x = windowSize.width + scrollOffset.x - paletteSize.width;
-    }
-  }
-  Blockly.WidgetDiv.position(xy.x, xy.y, windowSize, scrollOffset,
-                             this.sourceBlock_.RTL);
+  var saturationElements = this.createLabelDom_('Saturation');
+  div.appendChild(saturationElements[0]);
+  this.saturationReadout_ = saturationElements[1];
+  this.saturationSlider_ = new goog.ui.Slider();
+  this.saturationSlider_.setUnitIncrement(0.01);
+  this.saturationSlider_.setStep(0.001);
+  this.saturationSlider_.setMinimum(0.01);
+  this.saturationSlider_.setMaximum(0.99);
+  this.saturationSlider_.render(div);
+
+  var brightnessElements = this.createLabelDom_('Brightness');
+  div.appendChild(brightnessElements[0]);
+  this.brightnessReadout_ = brightnessElements[1];
+  this.brightnessSlider_ = new goog.ui.Slider();
+  this.brightnessSlider_.setUnitIncrement(2);
+  this.brightnessSlider_.setMinimum(5);
+  this.brightnessSlider_.setMaximum(255);
+  this.brightnessSlider_.render(div);
+
+  Blockly.FieldColour.hueChangeEventKey_ = goog.events.listen(this.hueSlider_,
```

Comment:
This might be fine given the changeListener is handled by closure, but I would carefully test, especially on touch devices, what happens when you start a slider change inside the editor then drag your mouse/finger outside of the element and mouse up somewhere else in the blockly window.